test/test_utils/test_reporting.py

```diff
@@ -0,0 +1,290 @@
+import csv
+import datetime
+import os
+import re
+
+from invoke.context import Context
+
+from test.test_utils import LOGGER
+from test.test_utils.ec2 import get_instance_num_gpus
+
+
+def get_test_coverage_file_path():
+    cwd = os.getcwd()
+
+    dlc_dir = cwd.split('/test/')[0]
+
+    return os.path.join(
+        dlc_dir,
+        "test",
+        f"test_coverage_report-{datetime.datetime.now().strftime('%Y-%m-%d')}.csv",
+    )
+
+
+def get_marker_arg_value(item_obj, marker_name, default=None):
+    """
+    Function to return the argument value of a pytest marker -- if it does not exist, fall back to a default.
+    If the default does not exist and the option does not exist, raise an error.
+    :param item_obj: pytest item object
+    :param marker_name: name of the pytest marker
+    :param default: default return value -- if None, assume this is a required marker
+    :return: First arg value for the marker or the default value
+    """
+    markers = [mark for mark in item_obj.iter_markers(name=marker_name)]
+    if not markers:
+        if not default:
+            raise RequiredMarkerNotFound(f"PyTest Marker {marker_name} is required on function {item_obj.name}")
+        return default
+    else:
+        return markers[0].args[0]
+
+
+class TestReportGenerator:
+    """
+    Class to generate test report files
+    """
+
+    ALLOWED_SINGLE_GPU_TESTS = ("telemetry", "test_framework_version_gpu")
+    SM_REPOS = (
+        "pytorch-training",
+        "pytorch-inference",
+        "tensorflow-tensorflow1_training",
```

Comment:
Why are these tensorflow-tensorflow1_training?

Reply:
We have separate tf1 and tf2 training repos for SageMaker. I think this might be more clear if I make these paths vs dash separated strings